src/deep_approx_strict_equal.js

```diff
@@ -0,0 +1,118 @@
+import approxStrictEqual from "./approx_strict_equal";
+import epsilon from "./epsilon";
+
+/**
+ * Approximate strict equality for all elements of two structures, which may be
+ * null, undefined, number, string, array, or object.  This implements a subset
+ * of the Node sourc file 'lib/internal/util/comparisons.js', which is mirrored
+ * by 'internal/util/comparisons.js' in the CommonJS assert module.
+ *
+ * @param {number} actual The value to be tested.
+ * @param {number} expected The reference value.
+ * @param {number} tolerance The acceptable relative difference between elements.
+ * @return {boolean} Whether all numbers are within tolerance.
+ */
+function deepApproxStrictEqual(actual, expected, tolerance = epsilon) {
+    // Types must match
+    if (typeof expected !== typeof actual) {
+        return false;
+    }
+
+    // Undefined matches undefined
+    else if (expected === undefined) {
+        return actual === undefined;
+    }
+
+    // Null only equals null (check both sides to prevent attempts to access
+    // properties of null further down)
+    else if (expected === null) {
+        return actual === null;
+    }
+    else if (actual === null) {
+        return false;
+    }
+
+    // Booleans must be exactly equal
+    else if (typeof expected === "boolean") {
+        return expected === actual;
+    }
+
+    // Strings must be exactly equal
+    else if (typeof expected === "string") {
+        return expected === actual;
+    }
+
```

Comment:
I think that the expansion to other types makes the meaning of this function a little weaker: this is only approximate for numbers, but strict for everything else. Especially if we enforce a strict TypeScript definition (only numbers and maybe arrays of numbers), and strict documentation, there's little reason to worry about people accidentally using this method for strings or any kind of unknown datatype: they're using a custom approximate equal method on purpose, and if they want another type of equality, they have implementations for that close at hand with the  `assert` module or the `===` operator.